CGAT/scripts/runGSEA.py

```diff
@@ -0,0 +1,1619 @@
+    f.write("@ " + str(sum(fdr_downregulated < 0.01)) +
+            " gene sets are significant at FDR < 1%" + "\n")
+    f.write("@ " +
+            str(sum(w2 < 0.01)) +
+            " gene sets are significantly enriched at nominal pvalue < 1%" +
+            "\n")
+    f.write("@ " +
+            str(sum(w2 < 0.05)) +
+            " gene sets are significantly enriched at nominal pvalue < 5%" +
+            "\n\n")
+    f.close()
+
+    # Plot enrichment score of top gene set for each phenotype.
+
+    # If specified number of top genesets for plotting enrichemnt score is higher than the total number of genesets.I will
+    # plot enrichemnt score for all genesets.
+    if(options.plot_no >= len(GG)):
+        options.plot_no = len(GG)
+    xcc = int(options.plot_no / 2) + 1
+    for i in range(0, options.plot_no):
+        if(len(nes_up_index) > 0):
+            AW = up_for_plot[nes_up_index[i]]
+            plot_enrichment_score(
+                store_enrichment_score,
+                original_es_index,
+                original_es,
+                in_list,
+                AW,
+                len(id))
+            plot_random_ES(store_permute, AW, in_list)
+
+        # Downregulated
+        if(len(nes_down_index) > 0):
+            AW = down_for_plot[nes_down_index[i]]
+            plot_enrichment_score(
+                store_enrichment_score,
+                original_es_index,
+                original_es,
+                in_list,
+                AW,
+                len(id))
+            plot_random_ES(store_permute, AW, in_list)
+    for i in range(0, options.plot_no):
+        if(len(nes_up_index) > 0):
+            AW = up_for_plot[nes_up_index[i]]
+            plot_enrichment_score_subplot(
+                store_enrichment_score,
+                original_es_index,
+                original_es,
+                in_list,
+                AW,
+                len(id),
+                xcc,
+                i + 1,
+                "enplot_upregulated_summary.jpeg",
+                options.plot_no)
+    for i in range(0, options.plot_no):
+        if(len(nes_down_index) > 0):
+            AW = down_for_plot[nes_down_index[i]]
+            plot_enrichment_score_subplot(
+                store_enrichment_score,
+                original_es_index,
+                original_es,
+                in_list,
+                AW,
+                len(id),
+                xcc,
+                i + 1,
+                "enplot_downregulated_summary.jpeg",
+                options.plot_no)
+    # Plot summary of top 20 genesets of each phenotype
+    plot_summary_report(
+        nes_up_index,
+        original_nes,
+        up_for_plot,
+        in_list,
+        "upregulated")
+    plot_summary_report(
+        nes_down_index,
+        original_nes,
+        down_for_plot,
+        in_list,
+        "downregulated")
+
+    # Generate pvalue vs ES graph.
+    AW = down_for_plot[nes_down_index]
```

Comment:
I would use lowercase for variables and give them meaningful names so it's easier to follow the code.